tests/e2e/envfuncsext/nebula-command.go

```diff
@@ -0,0 +1,164 @@
+package envfuncsext
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	nebulaclient "github.com/vesoft-inc/nebula-go/v3"
+	appspkg "github.com/vesoft-inc/nebula-operator/pkg/kube"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+type (
+	NebulaCommandOptions struct {
+		ClusterName      string
+		ClusterNamespace string
+		Username         string
+		Password         string
+		Space            string
+	}
+
+	nebulaCommandCtxKey struct {
+		clusterNamespace string
+		clusterName      string
+	}
+
+	NebulaCommandCtxValue struct {
+		Results nebulaclient.ResultSet
+	}
+)
+
+func GetNebulaCommandCtxValue(clusterNamespace, clusterName string, ctx context.Context) *NebulaCommandCtxValue {
+	v := ctx.Value(nebulaCommandCtxKey{
+		clusterNamespace: clusterNamespace,
+		clusterName:      clusterName,
+	})
+	data, _ := v.(*NebulaCommandCtxValue)
+	return data
+}
+
+func RunNGCommand(cmdCtx NebulaCommandOptions, cmd string) env.Func {
+	return func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+		client, err := appspkg.NewClientSet(cfg.Client().RESTConfig())
+		if err != nil {
+			return ctx, fmt.Errorf("error getting kube clientset: %v", err)
+		}
+
+		nodes, err := client.Node().ListAllNodes()
+		if err != nil {
+			return ctx, fmt.Errorf("error listing nodes in the cluster: %v", err)
+		}
+		if len(nodes) == 0 {
+			return ctx, fmt.Errorf("error listing nodes in the cluster: node list is empty")
+		}
+
+		service, err := client.Service().GetService(cmdCtx.ClusterNamespace, fmt.Sprintf("%v-graphd-svc", cmdCtx.ClusterName))
+		if err != nil {
+			return ctx, fmt.Errorf("error getting graphd service: %v", err)
+		}
+
+		ports := service.Spec.Ports
+
+		var portNum int
+		for _, port := range ports {
+			if port.Name == "thrift" {
+				portNum = int(port.NodePort)
+			}
+		}
+
+		if portNum <= 0 {
+			return ctx, fmt.Errorf("error getting service port: %v is <= 0", portNum)
+		}
+
+		hostAddress := nebulaclient.HostAddress{Host: nodes[0].Status.Addresses[0].Address, Port: portNum}
```

Comment:
It is not appropriate to connect to node. Maybe this address is unreachable.

Reply:
Got it, will fix this

Reply:
Fixed